src/framework/components/element/component.js

```diff
@@ -56,8 +56,10 @@ pc.extend(pc, function () {
      * @property {Number} right The distance from the right edge of the anchor. Can be used in combination with a split anchor to make the component's right edge always be 'right' units away from the right.
```

Comment:
Apologies for the noise in this file – it was necessary in order to introduce the new `calculatedWidth` and `calculatedHeight` properties. You can see most of those changes in isolation [here](https://github.com/playcanvas/engine/compare/master...calculated-dimensions) – I was planning to put them in as a separate PR, but in the end I figured it'd be more straightforward to test both sets of changes together. Can split it out if you don't agree.